CMakeLists.txt

```diff
@@ -91,6 +93,9 @@ option(NGRAPH_DISTRIBUTED_ENABLE "Add distributed mode to the CPU backend" FALSE
 option(NGRAPH_DEBUG_ENABLE "Enable output for NGRAPH_DEBUG statements" FALSE)
 option(NGRAPH_ONNX_IMPORT_ENABLE "Enable ONNX importer" FALSE)
 
+cmake_dependent_option(NGRAPH_DEX_ONLY "Remove codegen from the build" FALSE
```

Comment:
DEX does not depend on Intel or Nvidia GPU. They are independent. We should not limit the options based on DEX_ONLY. I would rename DEX_ONLY to something more descriptive to what it actually is.